src/virtualdom/items/Section/prototype/setValue.js

```diff
@@ -88,15 +88,19 @@ function reevaluateSection ( section, value ) {
 		owner:      section
 	};
 
+	section.hasContext = true;
+
 	// If we already know the section type, great
 	// TODO can this be optimised? i.e. pick an reevaluateSection function during init
 	// and avoid doing this each time?
 	if ( section.subtype ) {
 		switch ( section.subtype ) {
 			case SECTION_IF:
+			section.hasContext = false;
```

Comment:
Github won't let me add a line comment that's not a changed line, so [here's](https://github.com/ractivejs/ractive/blob/dev/src/virtualdom/items/Section/prototype/setValue.js#L128-L141) the block I'm referring to. It only gets to the reevaAsConditional part _if_ the value is not an object.